### DISCUSSION TITLE
Execute Appium test

### DISCUSSION THREAD

**Discussion by DavidAsulin1** [microsoft/HydraLab#570] [Q&A]
Hello! First, I want to say that this project is very cool! I enjoyed going through the source code.

Q: About Appium testing on a Mac with an iOS device, I'm a bit confused about how to obtain the device ID. The Appium flow isn't clear to me. I'd appreciate it if someone could help clarify things for me. Thanks!

**Comment by hydraxman** [comment]
Running Appium iOS on Mac is a new territory for us, we haven't tested it yet. Have you followed [this doc](https://github.com/microsoft/HydraLab/wiki/Test-agent-setup-on-MacOS)? Please let us know what detail issue you have encountered.

**Reply by DavidAsulin1** [reply]
@hydraxman I'm actually asking in general. Is the agent the user machine where the Appium script is executing? So, the only thing that Hydra benefits from in this case is the report in the center?

**Reply by hydraxman** [reply]
Yes, the Appium test cases run on the Hydra Lab agent machine.

And regarding the benefits, we think we are offering more than that, see this blog: https://medium.com/microsoft-mobile-engineering/how-hydra-lab-empowers-microsoft-mobile-testing-e4bd831ecf41

It's a cloud testing framework so it enables you to do test device management, task management, AI integration, etc.
